File: app/src/main/java/com/google/android/stardroid/util/Toaster.java
package com.google.android.stardroid.util;

import android.content.Context;
import android.widget.Toast;

import javax.inject.Inject;

/**
 * A wrapper around the Toast mechanism for easier unit testing.
 *
 * Created by johntaylor on 4/24/16.
 */
public class Toaster {
  private Context context;

  @Inject
  public Toaster(Context context) {
    this.context = context;
  }

  public void toastLong(int resId) {
    Toast.makeText(context, resId, Toast.LENGTH_LONG).show();
  }

  public void toastLong(String s) {
    Toast.makeText(context, s, Toast.LENGTH_LONG).show();
  }
}

File: app/src/main/java/com/google/android/stardroid/util/Toaster.kt
package com.google.android.stardroid.util

import android.content.Context
import javax.inject.Inject
import android.widget.Toast

/**
 * A wrapper around the Toast mechanism for easier unit testing.
 *
 * Created by johntaylor on 4/24/16.
 */
class Toaster @Inject constructor(private val context: Context) {
  fun toastLong(resId: Int) {
    Toast.makeText(context, resId, Toast.LENGTH_LONG).show()
  }

  fun toastLong(s: String?) {
    Toast.makeText(context, s, Toast.LENGTH_LONG).show()
  }
}